org.eclipse.richbeans.widgets/src/org/eclipse/richbeans/widgets/scalebox/NumberBox.java

```diff
@@ -781,8 +781,10 @@ public void setDecimalPlaces(int decimalPlaces) {
 	 */
 	@SuppressWarnings("unchecked")
 	public double getMaximum() {
-		if (maxProvider != null)
-			return maxProvider.getBoundValue();
+		if (maxProvider != null) {
+			double dbl = maxProvider.getBoundValue();
+			if (!Double.isNaN(dbl)) return dbl;
```

Comment:
Should this throw if db1 is null? Presumably that is the case where a maxProvider is set but it has given a null bound?